src/rshim.c

```diff
@@ -2773,21 +2777,30 @@ bool rshim_allow_device(const char *devname)
   return true;
 }
 
-static void *rshim_stop_thread(void *arg)
+/* Force to kill if not able to cleanup in time. */
+static void *rshim_force_stop(void *arg)
 {
-  /* Force to kill if not able to cleanup in time. */
   sleep(3);
   kill(0, SIGKILL);
```

Comment:
Maybe adding a logging here to highlight the abnormal situation, aka, the main thread hasn't exited normally, and we are resorting to forcibly kill our own process?